src/core/pyproxy.js

```diff
@@ -719,7 +719,7 @@ TEMP_EMJS_HELPER(() => {0, /* Magic, see comment */
     getBuffer : function(type = "u8") {
       let ArrayType = undefined;
       if (type) {
-        let ArrayType = type_to_array_map.get(type);
+        ArrayType = type_to_array_map.get(type);
```

Comment:
This is the one major bug fixed in this PR.

Reply:
Probably we should add some tests of `getBuffer` that actually use the `type` argument. Actually this will probably break the tests because there are a pair of bugs here: 
1. I meant for `type` to default to `undefined` but 
2. the type argument is always ignored.

Reply:
Might be good to in a different PR